### ISSUE TITLE
Examples in compose and concatenate schemas fail with new modeling

### ISSUE THREAD

**Issue by jdavies-st** [spacetelescope/asdf#687]
It seems some schemas don't work with the new code in `astropy.modeling`.

See the error log here:

https://travis-ci.org/spacetelescope/asdf/builds/563063710
```
Results (35.06s):
     673 passed
       6 failed
         - asdf-standard/schemas/stsci.edu/asdf/transform/compose-1.0.0.yaml:? 
         - asdf-standard/schemas/stsci.edu/asdf/transform/compose-1.1.0.yaml:? 
         - asdf-standard/schemas/stsci.edu/asdf/transform/compose-1.2.0.yaml:? 
         - asdf-standard/schemas/stsci.edu/asdf/transform/concatenate-1.0.0.yaml:? 
         - asdf-standard/schemas/stsci.edu/asdf/transform/concatenate-1.1.0.yaml:? 
         - asdf-standard/schemas/stsci.edu/asdf/transform/concatenate-1.2.0.yaml:? 
```

**Comment by perrygreenfield** [comment]
@jdavies-st can you point me to the specific tests that cause this? I don't find it terribly obvious from the error log.

**Comment by nden** [comment]
As far as I can tell these failures come from the schema examples, specifically the `compose` schema. I doubt these are failures in the code (because gwcs and jwst tests pass).

**Comment by nden** [comment]
Should be fixed when astropy/astropy#9047 is merged.

**Comment by nden** [comment]
Tests pass.
